python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,529 @@
+            "tvm.pipeline_executor.create", allow_missing=False
+        )
+        assert pipeline_create
+        module = pipeline_create(mods, config)
+
+        self.module_ = module
+
+    def graph_executor_create(self, pipeline_mods, mod_config):
+        """Create graph_executor list and return configuration as a json string.
+
+        Parameters
+        ----------
+        pipeline_mods : List[GraphExecutorFactoryModule]
+          List of GraphExecutorFactoryModule
+
+        mod_config : Dict[str, Any]
+            Modules dependency configuration information.
+
+        Returns
+        -------
+        mods : List[Module]
+            Module list.
+
+        mod_config : str
+            Mods configuration.
+        """
+
+        mods = []
+        for pipeline_mod in pipeline_mods:
+            mod = graph_executor.GraphModule(
+                pipeline_mod["default"](pipeline_mods[pipeline_mod]["dev"])
+            )
+            mods.append(mod.module)
+
+        return mods, json.dumps(mod_config)
+
+
+class PipelineConfig(object):
+    """The wrapper of each module to be pipelined. The wrapper mainly includes the
+    module itself as well as the binding that represents the connections of this
+    module's inputs and outputs to other modules.
+    """
+
+    class Binding:
+        """This class define the module connection information.
+        The type can only be either "input" or "output".
+
+        Parameters
+        ----------
+        owner : ModuleWrapper
+            The class who owns this interface.
+
+        io_type : str
+            The type of this interface. It can only be either "input" or "output".
+
+        name : str/integer
+            Name, for input it is string such as "data0", for output it is the
+            idx integer such as 0.
+        """
+
+        def __init__(self, owner, io_type, name, data_type=None):
+            self.io_owner = owner
+            self.io_type = io_type
+            self.name = str(name)
+            # Child nodes.
+            self.bindings = []
+            # Parents nodes.
+            self.parents = []
+
+            self.data_type = data_type
+
+        def get_name(self):
+            """Return the interface name and name of owner who own this interface."""
+            owner_name = ""
+            if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                owner_name = self.io_owner.name
+
+            return owner_name, self.name
+
+        def get_owner_idx(self):
+            """Return owner idex if owner is ModuleWrapper, if not return 0."""
+            if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                return self.io_owner.idx
+
+            # If not ModuleWrapper then owner is PipelineConfig, return 0
+            # to identify this is global interface
+            return 0
+
+        def is_global_interface(self):
+            """It is to check whether this interface is global interface."""
```

Comment:
This comment is not useful. What is global interface?

Reply:
pipeline executor have 2 types input/output, first is global interface that caller use these interface to set input and get output, second is  interface of internal subgraph module that build a pipeline from top to down.